aiida_mlip/data/model.py

```diff
@@ -231,32 +190,38 @@ def download(
         )
         arch_dir = (cache_dir / architecture) if architecture else cache_dir
 
-        # cache_path = cache_dir.resolve()
         arch_path = arch_dir.resolve()
         arch_path.mkdir(parents=True, exist_ok=True)
 
-        model_name = urlparse(url).path.split("/")[-1]
+        file = arch_path / filename
 
-        file = arch_path / filename if filename else arch_path / model_name
+        # Download file
+        request.urlretrieve(uri, file)
 
-        # If file already exists, use next indexed name
-        stem = file.stem
-        i = 1
-        while file.exists():
-            i += 1
-            file = file.with_stem(f"{stem}_{i}")
+        model = cls.from_local(file=file, architecture=architecture)
 
-        # Download file
-        request.urlretrieve(url, file)
+        if keep_file:
+            return model
 
-        if force_download:
-            print(f"filename changed to {file}")
-            return cls.local_file(file=file, architecture=architecture)
+        file.unlink(missing_ok=True)
+
+        qb = QueryBuilder()
```

Comment:
```suggestion
        # Check if the same model was used previously
        qb = QueryBuilder()
```
Did you mean to lose the comments here? It doesn't need to be line by line, but maybe this is enough?

Reply:
prob cancelled by mistake when I changed the things